### ISSUE TITLE
init_fn bug in estimator._train_model_distributed

### ISSUE THREAD

**Issue by fengyang0317** [tensorflow/tensorflow#20642]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.8.0
- **Python version**: 2.7
- **CUDA/cuDNN version**: 7.0
- **GPU model and memory**: GTX 980M

### Describe the problem
There is a bug in https://github.com/tensorflow/tensorflow/blob/4e883df1eaf52b7c655365eb5f6ecd9e5fe37457/tensorflow/python/estimator/estimator.py#L1327.

init_fn should take two arguments. However, the init_fn obtained in https://github.com/tensorflow/tensorflow/blob/4e883df1eaf52b7c655365eb5f6ecd9e5fe37457/tensorflow/python/estimator/estimator.py#L1269 is a lambda function which only takes one argument.

Currently, I am using the following line as a workaround.
```
init_fn=init_fn and (lambda x, y: init_fn(y)))
```

**Comment by tanzhenyu** [comment]
1. shouldn't init_fn always take two arguments?
2. do you have a code snippet for verifying this?

**Comment by fengyang0317** [comment]
@tanzhenyu 
`
import tensorflow as tf
import numpy as np


def model_fn(features, labels, mode, params):

  def init_fn(scaffold, session):
    pass

  scaffold = tf.train.Scaffold(init_fn=init_fn)
  return tf.estimator.EstimatorSpec(
    mode=mode,
    loss=labels[0],
    train_op=tf.no_op(),
    scaffold=scaffold
  )


def input_fn():
  dataset = tf.data.Dataset.from_tensors((np.zeros(10), np.zeros(1)))
  dataset = dataset.repeat()
  return dataset


def main(_):
  distribution_strategy = tf.contrib.distribute.OneDeviceStrategy(
    "device:GPU:0")

  run_config = tf.estimator.RunConfig(train_distribute=distribution_strategy)

  estimator = tf.estimator.Estimator(
    model_fn=model_fn,
    config=run_config
  )

  estimator.train(input_fn=input_fn, steps=10)


if __name__ == '__main__':
  tf.app.run()
`

**Comment by angersson** [comment]
@ispirmustafa can you take a look at this?

**Comment by ispirmustafa** [comment]
@josh11b this is about grouping multiple scaffolds. Based on desired outcome, either scaffold._user_init_fn should be used or a lambda should be created which a dummy first argument. 
Could you please take a look?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ispirmustafa, @anj-s: It has been 20 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @ispirmustafa, @anj-s: It has been 35 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @anj-s: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @anj-s: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @anj-s: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @anj-s: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by guptapriya** [comment]
I think https://github.com/tensorflow/estimator/commit/1f478bf50e922202e6a378557878bd049d4c9007#diff-d934b9f2c2d9384e077e7ab45e001f69 seems like a fix for this. @fengyang0317 can you try?

**Comment by guptapriya** [comment]
Thanks for verifying!
